File: 1-tdd-challenge/fizz-buzz/python-pytest/fizzbuzz.py
def fizzbuzz(i):
  if i % 15 == 0:
    return('fizzbuzz')
  elif i % 3 == 0:
    return('fizz')
  elif i % 5 == 0:
    return('buzz')
  else:
    return(i)

def main():
   for i in range(1,101):
    print(fizzbuzz(i))

if __name__ == '__main__':
  main()

File: 1-tdd-challenge/fizz-buzz/java-junit/FizzBuzz.java




public class FizzBuzz {
	public String generateFizzBuzz(Integer data) {
		if(data % 5 == 0 && data % 3 == 0) return "FizzBuzz";
		if(data % 3 == 0) return "Fizz";
		if(data % 5 == 0) return "Buzz";
		
		return data.toString();
	}
	
	public String createFizzBuzzSequence(int limit) {
		String result = "";
		for(int i = 1; i <= limit; i++) {
			result+=generateFizzBuzz(i) + ",";
		}
		
		result = result.substring(0, result.length()-1);
		
		return result;
	}
}

